apis/v1alpha2/backendtlspolicy_types.go

```diff
@@ -0,0 +1,148 @@
+
+	// Status defines the current state of BackendTLSPolicy.
+	Status PolicyStatus `json:"status,omitempty"`
+}
+
+// +kubebuilder:object:root=true
+// BackendTLSPolicyList contains a list of BackendTLSPolicies
+type BackendTLSPolicyList struct {
+	metav1.TypeMeta `json:",inline"`
+	metav1.ListMeta `json:"metadata,omitempty"`
+	Items           []BackendTLSPolicy `json:"items"`
+}
+
+// BackendTLSPolicySpec defines the desired state of
+// BackendTLSPolicy.
+// Note: BackendTLSPolicy is a Direct Attached Policy only.
+//
+// Support: Extended
+type BackendTLSPolicySpec struct {
+	// TargetRef identifies an API object to apply the policy to.
+	// Only Services have Extended support. Implementations MAY support
+	// additional objects, with Implementation Specific support.
+	// Note that this config applies to the entire referenced resource
+	// by default, but this default may change in the future to provide
+	// a more granular application of the policy.
+	//
+	// Support: Extended for Kubernetes Service
+	//
+	// Support: Implementation-specific for any other resource
+	//
+	TargetRef PolicyTargetReferenceWithSectionName `json:"targetRef"`
+
+	// TLS contains backend TLS policy configuration.
+	TLS *BackendTLSPolicyConfig `json:"tls"`
+}
+
+// BackendTLSPolicyConfig contains backend TLS policy configuration.
+// +kubebuilder:validation:XValidation:message="must not contain both CertRefs and StandardCerts",rule="(has(self.certRefs) && size(self.certRefs) > 0 && has(self.standardCerts) && self.standardCerts != \"\")"
+// +kubebuilder:validation:XValidation:message="must specify either CertRefs or StandardCerts",rule="!(has(self.certRefs) && size(self.certRefs) > 0 || has(self.standardCerts) && self.standardCerts != \"\")"
+type BackendTLSPolicyConfig struct {
+	// CertRefs contains one or more references to
+	// Kubernetes objects that contain PEM-encoded TLS certificates,
+	// which are used to establish a TLS handshake between the Gateway
+	// and backend Pod.
+	//
+	// If CertRefs is empty or unspecified, then StandardCerts must
+	// be specified.  Only one of CertRefs or StandardCerts may be
+	// specified, not both.
+	//
+	// If CertRefs is empty or unspecified, then system trusted
+	// certificates should be used. If there are none, or the
+	// implementation doesn't define system trusted certificates,
+	// then a TLS connection must fail.
+	//
+	// References to a resource in a different namespace are
+	// invalid.
+	//
+	// A single CertRef to a Kubernetes ConfigMap kind has "Core"
+	// support.  Implementations MAY choose to support attaching
+	// multiple certificates to a backend, but this behavior is
+	// implementation-specific.
+	//
+	// Support: Core - An optional single reference to a Kubernetes
+	// ConfigMap, with the CA certificate in a key named `ca.crt`.
+	//
+	// Support: Implementation-specific (More than one reference, or other kinds
+	// of resources).
+	//
+	// +kubebuilder:validation:MaxItems=8
```

Comment:
nit: MinItems=1? otherwise you can say `certRefs=[]` (I think)

Reply:
`MinItems=1` makes this not optional though, so I think we need to stick with this. In any case, we can add this later if we need it.

Reply:
Note that GEP-1897 states "CertRefs contains one or more (up to 64) references to Kubernetes objects..." and should be updated accordingly.

Reply:
I think CEL validation should be able to prevent invalid inputs like this, but I've filled #2473 to ensure we have good test coverage of our validation here before releasing this.